src/sqlfluff/rules/L053.py

```diff
@@ -0,0 +1,75 @@
+"""Implementation of Rule L053."""
+from typing import Optional
+
+from sqlfluff.core.rules.base import BaseRule, LintResult, LintFix, RuleContext
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L053(BaseRule):
+    """Top-level statements should not be wrapped in brackets.
+
+    | **Anti-pattern**
+    | A top-level statement is wrapped in brackets.
+
+    .. code-block:: sql
+       :force:
+
+        (SELECT
+            foo
+        FROM bar)
+
+        -- This also applies to statements containing a sub-query.
+
+        (SELECT
+            foo
+        FROM (SELECT * FROM bar))
+
+    | **Best practice**
+    | Don't wrap top-level statements in brackets.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+
+        -- Likewise for statements containing a sub-query.
+
+        SELECT
+            foo
+        FROM (SELECT * FROM bar)
+    """
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        """Top-level statements should not be wrapped in brackets."""
+        # We only care about bracketed segements that are direct
+        # descendants of a top-level statement segment.
+        if context.dialect.name == "tsql":
+            top_level_parent_stack_types = ["file", "batch", "statement"]
+        else:
+            top_level_parent_stack_types = ["file", "statement"]
```

Comment:
Why do we need `"statement"` in here? Shouldn't the top level statement always be `"file"` (or `"batch"` for t-sql)?

Reply:
@tunetheweb basically we need to find the first bracketed segment and say are its parents segments what we expect to find if it's the at the top level (syntactically speaking).

so in non-tsql the parse tree looks like:
FileSegment
|_StatementSegment
   |_BracketedSegment

and for tsql the parse tree gives:
FileSegment
|_BatchSegment
   |_StatementSegment
      |_BracketedSegment

So I'm basically just looking at the parents and seeing if it's what we'd expect

Reply:
Ah I've got you. It's because the BracketedStatement is part of a statement right.

So does the code below, basically say we have to right up the tree (until we get to the end - the `FileSegment`), and if we find any other type (other than those three), then it's not a top level segment so find to have brackets?

```
        if not (
            context.segment.is_type("bracketed")
            and [segment.type for segment in context.parent_stack]
            == top_level_parent_stack_types
        ):
            return None
```

Reply:
Exactly right, although because of the list I'm being even more specific and saying I expect the top_level_parent_stack_types to be in the exact order and number as in the list i'm comparing too 😄

Reply:
As per [other conversation above](https://github.com/sqlfluff/sqlfluff/pull/1916#discussion_r751813853) that even more specific might be unnecessary. Resolving this, so we can continue conversation in that thread.